src/modules/direction/store/directionSlice.ts

```diff
@@ -161,6 +166,25 @@ export const directionsSlice = createSlice({
         direction.materials = action.payload.materials;
       }
     },
+    setPostFilters: (
+      state,
+      action: PayloadAction<{
+        key: DirectionFilterTypes;
+        filters: IFilter;
+        directionName: string;
+      }>,
+    ) => {
+      const { key, filters, directionName } = action.payload;
+      const direction = state[directionName] as IDirectionState;
+      if (direction) {
+        direction.filters = {
+          ...direction.filters,
+          [key]: filters,
+        };
+        direction.materials.meta.pageNumber = -1;
+        direction.materials.posts.length = 0;
```

Comment:
should we also reset the posts?

Reply:
yes, because now we have fetchInitialMaterials that checks if posts.length = 0